### PR TITLE
feat: Add option InjectAsBase64 for ARM

### PR DESCRIPTION
**Pull request by twinpiloot**
Unfortunately, I don't know how to run the test locally. Also don't know how to update the version number.

### CODE REVIEW

**Comment by stijnmoreels** [comment]
Hi Martin, great thank you for your addition and work on Arcus.
Couple of things:
- [ ] Please read the [contribution guide](https://github.com/arcus-azure/arcus.scripting/blob/main/CONTRIBUTING.md) to answer already couple of your questions;
- [ ] Please revert the changes in the v1.3.1 folder, it is not up to contributors to determine when we release. If changes are needed to the docs, please change them in the `/preview` folder;
- [ ] After that, we can start reviewing.

Thx again!

**Comment by twinpiloot** [comment]
I cannot run the tests locally for some reason:

PS C:\Users\twinp> Invoke-Pester C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Arcus.Scripting.ARM.tests.ps1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           WARNING: The names of some imported commands from the module 'Arcus.Scripting.ARM' include unapproved verbs that might  make them less discoverable. To find the commands with unapproved verbs, run the Import-Module command again with the   Verbose parameter. For a list of approved verbs, type Get-Verb.                                                         Describing Arcus ARM integration tests
   Context ARM injection
Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-inline-value.json' for injection tokens                                                                                      Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-inline.json' for injection tokens                                                                                            Finished script 'Inject-ArmContent.ps1' on path 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-inline.json'                                                                                                     [-] Replaces file path with inline file contents 773ms                                                                    RuntimeException: '-Be' is not a valid Should operator.                                                                 at Get-TestResult, C:\Program Files\WindowsPowerShell\Modules\Pester\3.4.0\Functions\Assertions\Should.ps1: line 42                                                                                                                             at <ScriptBlock>, C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Arcus.Scripting.ARM.tests.ps1: line 18                                                                                                                   Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-object-value (windows).json' for injection tokens

**Comment by stijnmoreels** [comment]
> I cannot run the tests locally for some reason:
>  RuntimeException: '-Be' is not a valid Should operator. at Get-TestResult, C:\Program Files\WindowsPowerShell\Modules\Pester\3.4.0\Functions\Assertions\Should.ps1: line 42 at , C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Arcus.Scripting.ARM.tests.ps1: line 18 

This shows a conflicting local PS Pester package version as there were breaking changes regarding the `-Be` operator in some of their versions.

**Comment by twinpiloot** [comment]
I was able to run the test locally after upgrading Pester to 5.6.1

**Comment by twinpiloot** [comment]
You can delete this PR. I'm giving it up. Takes me too much time. Test is succesful locally, but the same test fails in the pipeline.
Will find another solution.

PS C:\Windows\System32> Invoke-Pester C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Arcus.Scripting.ARM.tests.ps1

Starting discovery in 1 files.
WARNING: The names of some imported commands from the module 'Arcus.Scripting.ARM' include unapproved verbs that might
make them less discoverable. To find the commands with unapproved verbs, run the Import-Module command again with the
Verbose parameter. For a list of approved verbs, type Get-Verb.
Discovery found 5 tests in 1.99s.
Running tests.
Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-inline-value.json' for injection tokens
Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-inline.json' for injection tokens
Finished script 'Inject-ArmContent.ps1' on path 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-inline.json'
Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-object-value (windows).json' for injection tokens
WARNING: Content to inject into ARM template file
'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-object (windows).json' cannot be
parsed as a JSON object!
Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-object (windows).json' for injection tokens
Finished script 'Inject-ArmContent.ps1' on path 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-object (windows).json'
Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-object-value (windows).json' for injection tokens
WARNING: Content to inject into ARM template file
'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-object-absolutepath (windows).json'
cannot be parsed as a JSON object!
Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-object-absolutepath (windows).json' for injection tokens
Finished script 'Inject-ArmContent.ps1' on path 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-object-absolutepath (windows).json'
Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-escape-value.xml' for injection tokens
Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-escape.json' for injection tokens
Finished script 'Inject-ArmContent.ps1' on path 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-escape.json'
Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\codit.eu.crt' for injection tokens
Done checking ARM template file 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-certificate.json' for injection tokens
Finished script 'Inject-ArmContent.ps1' on path 'C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Files\arm-template-certificate.json'
[+] C:\Repos\arcus.scripting\src\Arcus.Scripting.Tests.Integration\Arcus.Scripting.ARM.tests.ps1 7.62s (2.09s|3.77s)
Tests completed in 7.78s
Tests Passed: 5, Failed: 0, Skipped: 0, Inconclusive: 0, NotRun: 0

**Comment by stijnmoreels** [comment]
> You can delete this PR. I'm giving it up. Takes me too much time. Test is succesful locally, but the same test fails in the pipeline. Will find another solution.

It's probably a different test remotely, as it succeeds remotely on windows but fails on linux (and I'm guessting that you work with a windows system locally).

Putting @pim-simons in CC.

**Comment by pim-simons** [comment]
Turns out base64 encoding on Linux returns a bit of a different string than on Windows. Decoding both base64 strings from Linux and Windows returns the same correct file so I dont really think it is an issue. Created seperate Windows and Linux tests and all working fine now 👍🏻

**Comment by twinpiloot** [comment]
Thanks Stijn and Pim for finding and fixing this.

**Review by pim-simons — APPROVED** [review]
Looks good to me 👍

**Comment by pim-simons** [comment]
> Thanks Stijn and Pim for finding and fixing this.

Thanks for contributing! 🥇

**Comment by pim-simons** [comment]
@twinpiloot PR doesn't have to be closed, will wait for @stijnmoreels  to review and then we can squash and merge to main 👍🏻

File: src/Arcus.Scripting.Tests.Integration/Files/codit.eu.crt
Comment (stijnmoreels) [inline src/Arcus.Scripting.Tests.Integration/Files/codit.eu.crt:1]: Is this some self-signed public certificate?

**Review by stijnmoreels — APPROVED** [review]
Fine for me as well.

**Comment by twinpiloot** [comment]
Its the public certificate of the codit website. Just used for the test



Verzonden vanaf mijn Galaxy



-------- Oorspronkelijk bericht --------
Van: Stijn Moreels ***@***.***>
Datum: 29-11-2024 21:19 (GMT+01:00)
Aan: "arcus-azure/arcus.scripting" ***@***.***>
Cc: Martin Peters ***@***.***>, Mention ***@***.***>
Onderwerp: Re: [arcus-azure/arcus.scripting] feat: Add option InjectAsBase64 for ARM (PR #438)


@stijnmoreels approved this pull request.

Fine for me as well.

________________________________

On src/Arcus.Scripting.Tests.Integration/Files/codit.eu.crt<https://github.com/arcus-azure/arcus.scripting/pull/438#discussion_r1863909719>:

Is this some self-signed public certificate?

**Pull request merged**